lib/install/validate/storage.go

```diff
@@ -39,29 +39,39 @@ func (v *Validator) storage(ctx context.Context, input *data.Data, conf *config.
 		v.NoteIssue(err)
 		return
 	}
+	if err != nil {
+		v.NoteIssue(err)
+	}
 
 	// provide a default path if only a DS name is provided
 	if imageDSpath.Path == "" {
 		imageDSpath.Path = input.DisplayName
 	}
 
-	v.NoteIssue(err)
 	if ds != nil {
 		v.SetDatastore(ds, imageDSpath)
 		conf.AddImageStore(imageDSpath)
 	}
 
+	if !v.targetDatastoreIsWriteable(ctx, ds) {
+		return
+	}
+
 	if conf.VolumeLocations == nil {
 		conf.VolumeLocations = make(map[string]*url.URL)
 	}
 
 	// TODO: add volume locations
 	for label, volDSpath := range input.VolumeLocations {
-		dsURL, _, err := v.DatastoreHelper(ctx, volDSpath, label, "--volume-store")
+		dsURL, vds, err := v.DatastoreHelper(ctx, volDSpath, label, "--volume-store")
```

Comment:
It's not enough to check if there is one host connected to this datastore, cause if two datastores connected two different hosts, vch deployment will have problem as well.
So you should do the checking after all datastore validation is finished.